python/tk_houdini/base_hooks/node_handler_base.py

```diff
@@ -0,0 +1,471 @@
+            script_callback=self.generate_callback_script_str("enable_sgtk"),
+            script_callback_language=hou.scriptLanguage.Python
+        )
+        sgtk_templates.append(use_sgtk)
+
+        sgtk_templates.extend(self._create_sgtk_parms(node))
+
+        all_versions = hou.StringParmTemplate(
+            self.SGTK_ALL_VERSIONS,
+            "All Versions",
+            1,
+            is_hidden=True
+        )
+        sgtk_templates.append(all_versions)
+
+        version = hou.MenuParmTemplate(
+            self.SGTK_VERSION,
+            "Version",
+            tuple(),
+            item_generator_script=self.generate_callback_script_str("populate_versions"),
+            item_generator_script_language=hou.scriptLanguage.Python,
+            script_callback=self.generate_callback_script_str("refresh_file_path_from_version"),
+            script_callback_language=hou.scriptLanguage.Python
+        )
+        version.setConditional(hou.parmCondType.DisableWhen, "{ use_sgtk != 1 }")
+        version.setJoinWithNext(True)
+        sgtk_templates.append(version)
+
+        refresh_button = hou.ButtonParmTemplate(
+            self.SGTK_REFRESH_VERSIONS,
+            "Refresh Versions",
+            script_callback=self.generate_callback_script_str("refresh_file_path_from_version"),
+            script_callback_language=hou.scriptLanguage.Python
+        )
+        refresh_button.setConditional(
+            hou.parmCondType.DisableWhen,
+            '{ use_sgtk != 1 } { sgtk_element == "" }'
+        )
+        refresh_button.setJoinWithNext(True)
+        sgtk_templates.append(refresh_button)
+
+        resolved_version = hou.StringParmTemplate(
+            self.SGTK_RESOLVED_VERSION,
+            "Resolved Version",
+            1,
+            default_value=("1",)
+        )
+        resolved_version.setConditional(hou.parmCondType.DisableWhen, "{ sgtk_version != -1 }")
+        sgtk_templates.append(resolved_version)
+        
+        engine_version = self.parent.version if self.parent.version != "Undefined" else "DEV"
+        sgtk_folder = hou.FolderParmTemplate(
+            self.SGTK_FOLDER,
+            "SGTK (ver: {})".format(engine_version),
+            parm_templates=sgtk_templates,
+            folder_type=hou.folderType.Simple
+        )
+        return sgtk_folder
+
+    def _set_up_node(self, node, parameter_group, hou=None):
+        """
+        Set up a node for use with shotgun pipeline.
+
+        :param node: A :class:`hou.Node` instance.
+        :param parameter_group: A :class:`ParmGroup` instance.
+        :param hou: The houdini module. We have to lazy load the houdini python
+            module here, but not in the hooks, so use hook's imports and pass it
+            for efficiency.
+        """
+        if not hou:
+            import hou
+        self._set_up_parms(node)
+        sgtk_folder = self._create_sgtk_folder(node)
+        if not self.SGTK_IDENTIFIER in parameter_group:
```

Comment:
Coding standard suggestion from flake8: test for membership should be 'not in'
```suggestion
        if self.SGTK_IDENTIFIER not in parameter_group:
```